convert sign in web app guide to single page

chrisdmills-okta opened the pull request:
<!-- PLEASE REMEMBER THAT THIS A PUBLIC REPO AND ANY PR AND SUBSEQUENT DISCUSSION WILL BE VISIBLE TO ANYONE AND EVERYONE -->

## Description:
- **What's changed?** https://developer.okta.com/docs/guides/sign-into-web-app/aspnet/before-you-begin/ guide updated to single page 
- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-438731](https://oktainc.atlassian.net/browse/OKTA-438731)

brianduffield-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-web-app/main/index.md:
```diff
@@ -0,0 +1,178 @@
+
+**Sample code**
+
+* ASP.NET — see our [ASP.NET MVC example](https://github.com/okta/samples-aspnet/tree/master/okta-hosted-login) for a working sample app, or check out [ASP.NET MVC quickstart](https://docs.microsoft.com/en-us/aspnet/mvc/overview/getting-started/introduction/getting-started) if you want to build your own to work through this guide with.
```
Our style is to use em-dashes here, not hypens; applies to all bullets

chrisdmills-okta replied:
Those were literal em-dash characters, created using alt + shift + - on the mac keyboard, but I appreciate it is easier to reason with if you use em-dash HTML entity refs. I've changed them, and will use them in future.

chrisdmills-okta replied:
those were literal em-dash characters (created using Alt + Shift + - on mac keyboard), but I appreciate HTML entities are probably easier to reason with, so I've changed to those

brianduffield-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-web-app/main/index.md:
```diff
@@ -0,0 +1,178 @@
+1. Scroll to the **General Settings** panel, and click **Edit**.
+1. In the **Allowed grant types**, select **Refresh Token**.
+1. In the **Refresh Token** section, select **Rotate token after every use**.
+    > **Note:** The default number of seconds for the **Grace period for token rotation** is set to 30 seconds. You can [change the value](/docs/guides/refresh-tokens/refresh-token-rotation/#enable-refresh-token-rotation) to any number between 0 and 60 seconds. After the refresh token is rotated, the previous token remains valid for this amount of time to allow clients to get the new token. Using ...
```
When built, there's no space between step and the note.

chrisdmills-okta replied:
I've added a space, and also deindented it so that it appears at the same indent level as the note further below.

chrisdmills-okta replied:
added

brianduffield-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-web-app/main/index.md:
```diff
@@ -0,0 +1,178 @@
+
+When a user signs in, their profile information (stored in Okta) is made available to your application. It's common to use this info to update your application's UI.
+
+By default, the profile items (called "claims") that Okta returns include the user's email address, name and preferred username. 
```
add serial comma

chrisdmills-okta replied:
added

chrisdmills-okta replied:
added

brianduffield-okta reviewed (changes requested):
A few nits and a couple broken links:

https://developer.okta.com/docs/guides/ (Step 2)
https://developer.okta.com/docs/guides/sharing-cert/main/ (under what you need)

brianduffield-okta commented:
Still need to review conductor.yml file

chrisdmills-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-web-app/main/index.md:
```diff
@@ -0,0 +1,178 @@
+**Learning outcomes**
+
+* Create an Okta app integration to represent your app in your org.
+* Add dependencies and configure your app ready to use Okta for authentication.
```
I don't think we do; removed!

chrisdmills-okta replied:
Nope; removed!

chrisdmills-okta commented:
Thanks for the comments so far @brianduffield-okta . Nits and broken links fixed.

brianduffield-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -2793,3 +2793,51 @@ redirects:
     to: /docs/guides/configure-signon-policy/main/#prompt-for-an-mfa-factor-when-a-user-is-outside-the-us
   - from: /docs/guides/configure-signon-policy/next-steps/index.html
     to: /docs/guides/configure-signon-policy/main/#next-steps
+  - from: /guides/sign-into-web-app/-/before-you-begin/index.html
```
I believe you purposely left out the language-dependent stack selector from the "before-you-begin" section (and added that content generically under the Nutrition facts Sample heading), so, we don't need to have the hyphen language selector in the re-direct, nor can we reference "before-you-begin" in the To: section because that heading no longer exists. This re-direct can simply go to the top of the single page. I.e., the two re-directs should look like this:

`from: /docs/guides/sign-into-web-app/-/before-you-begin/index.html`
`to: /docs/guides/sign-into-web-app/`
`from: /docs/guides/sign-into-web-app/-/before-you-begin`
`to: /docs/guides/sign-into-web-app/`

brianduffield-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -2793,3 +2793,51 @@ redirects:
+  - from: /guides/sign-into-web-app/-/redirect-to-sign-in/index.html
+    to: /docs/guides/sign-into-web-app/-/redirect-to-sign-in/
+  - from: /guides/sign-into-web-app/-/require-authentication/index.html
+    to: /docs/guides/sign-into-web-app/-/require-authentication/
```
It appears all the "to:" redirects are incorrect. To go to a heading within the single page, we need to add the anchor. For example:

`from: /docs/guides/sign-into-web-app/-/define-callback/index.html`
`to: /docs/guides/sign-into-web-app/main/#understand-the-callback-route`
`from: /docs/guides/sign-into-web-app/-/define-callback`
`to: /docs/guides/sign-into-web-app/main/#understand-the-callback-route`

vanngo-okta replied:
You need the generic stack selector in the `to` clause (if there are stack selectors): 
```
from: /docs/guides/sign-into-web-app/-/define-callback/index.html
to: /docs/guides/sign-into-web-app/-/main/#understand-the-callback-route
from: /docs/guides/sign-into-web-app/-/define-callback
to: /docs/guides/sign-into-web-app/-/main/#understand-the-callback-route
```

brianduffield-okta replied:
Updated my example above as per Van's comment. Thanks, Van!

brianduffield-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -2793,3 +2793,51 @@ redirects:
+    to: /docs/guides/sign-into-web-app/aspnet/define-callback/
+  - from: /guides/sign-into-web-app/aspnet/get-user-info/index.html
+    to: /docs/guides/sign-into-web-app/aspnet/get-user-info/
+  - from: /guides/sign-into-web-app/aspnet/next-steps/index.html
```
No need to hard-code the languages in the redirect, the hyphen suffices.

brianduffield-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -2793,3 +2793,51 @@ redirects:
+  - from: /guides/sign-into-web-app/springboot/redirect-to-sign-in/index.html
+    to: /docs/guides/sign-into-web-app/springboot/redirect-to-sign-in/
+  - from: /guides/sign-into-web-app/springboot/require-authentication/index.html
+    to: /docs/guides/sign-into-web-app/springboot/require-authentication/
```
There are numerous existing redirects that need to be updated as well, see lines: 1241-1287 and 1502-1549 and elsewhere. (A lot of those original re-directs were incorrectly hard-coded as well; we might be able to reduce some lines). There's also a sprinkling of other lines, too. Do a search on "sign-into-web-app" in the conductor file.

chrisdmills-okta commented:
Thanks a lot for your guidance here, @brianduffield-okta and @vanngo-okta . I think this was one of the last ones I did before I actually had a clue what I was doing WRT the redirects; sorry about that (I submitted it very recently, but I copied it from an old PR that I had got into a mess on, so I decided to do a fresh one).

So for the redirects, I have:

1. Put the redirects in the same order as the guides and sections they redirect to appear in the guide. They were in a weird order, which I thought would make it even harder to figure out in future.
2. Redirected them to fragments, not separate URLs.
3. Included /docs/ at the start of all the from: ones.
4. Included /main/ in all the to: ones that need it
5. Included two copies of each, one with /index.html and one without.
6. Removed all the platform specific ones and just kept the neutral ones.
7. In the old redirects, removed all the language-specific ones, removed the ones that were duplicates, and made sure to redirect to the most modern URLs available (e.g. the fragment ones)

Let me know if you think this one needs anything else.

brianduffield-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -1238,53 +1238,21 @@ redirects:
   - from: /guides/sign-into-spa/vue/user-info/index.html
     to: /docs/guides/sign-into-spa/vue/user-info/
   - from: /guides/sign-into-web-app/-/before-you-begin/index.html
```
In theory, we should add the secondary From: redirect here (i.e. /guides/sign-into-web-app/-/before-you-begin) for all these old re-directs as they were not implemented properly originally. However, since these are old URLs and there've been no complaints//404 logs perhaps we're okay?

vanngo-okta replied:
I'm not sure if the old **from** was implemented incorrectly? What if the old-old structure didn't include `/docs/`?? At least that's what I'm assuming when I worked on the redirects. We can only assume that **from** is correct, and that we are controlling the **to** clause.

chrisdmills-okta replied:
Same here, this is what I assumed. We can't keep trying to second-guess fix the old redirects like this.

vanngo-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -1499,54 +1467,6 @@ redirects:
     to: /docs/guides/sign-into-spa/vue/use-the-access-token/
   - from: /guides/sign-into-spa/vue/user-info/index.html
     to: /docs/guides/sign-into-spa/vue/user-info/
-  - from: /guides/sign-into-web-app/-/before-you-begin/index.html
```
Was this here from before? Maybe we should leave it as-is? We shouldn't be removing previous redirects unless we know for sure that the **from** URL never existed before?

chrisdmills-okta replied:
I deleted this block of redirects because they were exactly the same as the previous block that I left in.

vanngo-okta commented on packages/@okta/vuepress-site/docs/guides/implement-grant-type/main/authcode/install-sdk.md:
```diff
@@ -2,4 +2,4 @@
 Okta recommends using existing libraries and OAuth 2.0 helper methods to implement your authentication flow. You can use one of Okta's many language and framework SDKs or an open-source library if an appropriate Okta SDK is not available. See [Languages & SDKs overview](/code/) for a list of Okta SDKs that you can download to start using with your app.
 
-Refer to [Add and configure packages](/docs/guides/sign-into-web-app/-/configure-packages) for instructions on how to install and use Okta back-end framework SDKs. You can download Okta sample apps to see how the SDKs are used in your app's framework. See [Examples](#examples) for a list of sample apps.
+Refer to [Add and configure packages](/docs/guides/sign-into-web-app/aspnet/main/#add-and-configure-packages) for instructions on how to install and use Okta back-end framework SDKs. You can download Okta sample apps to see how the SDKs are used in your app's framework. See [Examples](#examples) for a list of sample apps.
```
I think this is ok for now (hardcoding `aspnet` will avoid the left-nav bug). But in the future, if we implement the global language-selector for the site, all this hardcoding will be an issue.

chrisdmills-okta replied:
OK. I'm going to try to get the /-/ bug fixed asap by the agency I'm hiring, so hopefully that will be OK soonish. We can always do a search and replace to get rid of all the hardcoded URLs when the time comes.

chrisdmills-okta commented on packages/@okta/vuepress-site/conductor.yml:
```diff
@@ -1238,53 +1238,21 @@ redirects:
   - from: /guides/sign-into-spa/vue/user-info/index.html
     to: /docs/guides/sign-into-spa/vue/user-info/
```
I've got a separate PR dealing with the sign-into-spa guide: https://github.com/okta/okta-developer-docs/pull/2669/files I need to go through the historic redirects anyway in there, so I'll fix that over there.

chrisdmills-okta commented:
Thanks for the comments @vanngo-okta ; all answered. This being the case, are we OK to merge now?

vanngo-okta reviewed (approved):
Looks good! 🚢 
Just remember to test all the **from** redirects after the merge.
